### PR TITLE
mysql8: update to 8.0.26

### PR DESCRIPTION
**Pull request by chrstphrchvz**
#### Description
https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-26.html#mysqld-8-0-26-compiling (should also now support building on Apple Silicon)

###### Type(s)

- [x] bugfix
- [x] enhancement
- [x] security fix

###### ~~Tested on~~

~~macOS 10.15.7~~
~~Xcode 12.4 command line tools~~
In progress…

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: databases/mysql8/Portfile [hunk 1]
```diff
@@ -69,6 +70,7 @@ if {$subport eq $name} {
 
     use_parallel_build  yes
 
+    # Will require C++17 as of 8.0.27
```
Comment (chrstphrchvz) [inline databases/mysql8/Portfile:73]: https://dev.mysql.com/doc/refman/8.0/en/source-installation-prerequisites.html says:
> As of MySQL 8.0.27, MySQL 8.0 source code permits use of C++17 features.

so I have reminders to update `compiler.cxx_standard` to `2017` when updating to 8.0.27.

**Comment by chrstphrchvz** [comment]
https://github.com/macports/macports-ports/runs/3514137141#step:8:10534

```
[ 71%] Building CXX object sql/CMakeFiles/sql_gis.dir/gis/buffer.cc.o
cd /opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/build/sql && /usr/bin/clang++ -DBOOST_ALL_NO_LIB -DHAVE_CONFIG_H -DHAVE_TLSv13 -DLZ4_DISABLE_DEPRECATE_WARNINGS -DMYSQL_SERVER -DRAPIDJSON_NO_SIZETYPEDEFINE -DRAPIDJSON_SCHEMA_USE_INTERNALREGEX=0 -DRAPIDJSON_SCHEMA_USE_STDREGEX=1 -D_USE_MATH_DEFINES -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/build -I/opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/build/include -I/opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/mysql-8.0.26 -I/opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/mysql-8.0.26/include -I/opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/boost_1_73_0 -I/opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/mysql-8.0.26/include/boost_1_73_0/patches -isystem /opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/mysql-8.0.26/extra/rapidjson/include -isystem /opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/mysql-8.0.26/extra/protobuf/protobuf-3.11.4/src -isystem /opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/mysql-8.0.26/extra/lz4/lz4-1.9.3 -isystem /opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/mysql-8.0.26/extra/libedit/libedit-20191231-3.1/src/editline -isystem /opt/local/include -std=c++14 -fno-omit-frame-pointer -ftls-model=initial-exec -pipe -Os -DNDEBUG -I/opt/local/include -stdlib=libc++ -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -Wall -Wextra -Wformat-security -Wvla -Wundef -Wmissing-format-attribute -Woverloaded-virtual -Wcast-qual -Wno-null-conversion -Wno-unused-private-field -Wconditional-uninitialized -Wdeprecated -Wextra-semi -Wheader-hygiene -Wnon-virtual-dtor -Wundefined-reinterpret-cast -Winconsistent-missing-destructor-override -Winconsistent-missing-override -Wshadow-field -ffunction-sections -fdata-sections -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -mmacosx-version-min=10.14 -fPIC   -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -Wshadow-uncaptured-local -MD -MT sql/CMakeFiles/sql_gis.dir/gis/buffer.cc.o -MF CMakeFiles/sql_gis.dir/gis/buffer.cc.o.d -o CMakeFiles/sql_gis.dir/gis/buffer.cc.o -c /opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/mysql-8.0.26/sql/gis/buffer.cc
/opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/mysql-8.0.26/sql/gis/buffer.cc:70:7: error: no matching constructor for initialization of 'bg::strategy::buffer::geographic_point_circle<>'
      geo_point_circle(32, bg::srs::spheroid<double>(srs->semi_major_axis(),
      ^
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/var/macports/build/_Users_runner_work_1_s_ports_databases_mysql8/mysql8/work/boost_1_73_0/boost/geometry/strategies/geographic/buffer_point_circle.hpp:56:14: note: candidate constructor not viable: allows at most single argument 'count', but 2 arguments were provided
```

I have not found what’s causing this error. The desired constructor is in a header file that MySQL has patched, so my guess is that the patched header isn’t included properly or it is being “shadowed” by the unpatched header.

**Comment by herbygillot** [comment]
Thanks for looking at this @chrstphrchvz. There was a patch I had to add a while back to force `mysql8` to use its local Boost as it was using the external Boost lib for whatever reason.  Perhaps try disabling it.  The patch name is `patch-sql-local-boost.diff`.

**Comment by chrstphrchvz** [comment]
Thanks for pointing me to the patch. I think the
at may indicate some other issue still present, but for now I would guess `${BOOST_PATCHES_DIR}` needs to remain included before `${BOOST_INCLUDE_DIR}`, so I’ll give that a try.

**Comment by chrstphrchvz** [comment]
Tweaking the patch worked. If others believe this PR is ready, could a committer please ["squash and merge"](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request#merging-a-pull-request) it with the message:

```
mysql8: update to 8.0.26

Add reminder to set compiler.cxx_standard to 2017 when updating

${BOOST_PATCHES_DIR} needs to remain included before
${BOOST_INCLUDE_DIR} in patch-sql-local-boost.diff
```

**Comment by herbygillot** [comment]
Merged.  Thank you for getting this sorted, @chrstphrchvz.

**Comment by xeron** [comment]
Mysql keeps rebuilding in a loop after this update:

```
--->  Cleaning mysql8
--->  Updating database of binaries
--->  Scanning binaries for linking errors
--->  Found 1 broken file, matching files to ports
--->  Found 1 broken port, determining rebuild order
--->  Rebuilding in order
     mysql8 @8.0.26_0
```

**Comment by chrstphrchvz** [comment]
@xeron Does `port -v rev-upgrade` say which file is broken?

**Comment by xeron** [comment]
@chrstphrchvz 

```
$ sudo port -v rev-upgrade
--->  Scanning binaries for linking errors
Could not open /opt/local/lib/mysql8/plugin/libprotobuf-lite.3.11.4.dylib: Error opening or reading file (referenced from /opt/local/lib/mysql8/plugin/group_replication.so)
--->  Found 1 broken file, matching files to ports
--->  Found 1 broken port, determining rebuild order
You can always run 'port rev-upgrade' again to fix errors.
The following ports will be rebuilt: mysql8 @8.0.26
```

**Comment by chrstphrchvz** [comment]
It turns out my local build never completed due to an error that only happens when using trace mode: https://trac.macports.org/ticket/63455

I have yet to reproduce @xeron's issue (symbolic link seems to erroneously point to `../../mysql/…` instead of `../mysql/…`) or think of what's causing it; for now I would suggest opening a ticket and providing main.log for others to compare with.

**Comment by chrstphrchvz** [comment]
I have now reproduced the symbolic link issue and opened a ticket, please continue the discussion there:  https://trac.macports.org/ticket/63456

**Pull request merged**